runtime/zlib.cpp

```diff
@@ -58,13 +65,147 @@ const string_buffer *zlib_encode(const char *s, int32_t s_len, int32_t level, in
   return &static_SB;
 }
 
+class_instance<C$DeflateContext> f$deflate_init(int encoding, const array<mixed> &options) {
+  int level = -1;
+  int memory = 8;
+  int window = 15;
+  int strategy = Z_DEFAULT_STRATEGY;
+  string_buffer dict;
+  for (const auto &option : options) {
+    mixed value;
+    if (!option.is_string_key()) {
+      php_warning("deflate_init() : unsupported option");
+      return {};
+    }
+    if (option.get_string_key() == string("level")) {
+      value = options.get_value(string("level"));
```

Comment:
`option` is iterator, you can simply get value by this :) 
```suggestion
      value = option.get_value();
```